### ISSUE TITLE
Moving InternalStateType to a string based input

### ISSUE THREAD

**Issue by Micket** [oofem/oofem#1]
The enum that makes up IST's has numerous problems:
1. Statically defined; can't add new material model outputs without going upstream (which otherwise works thanks to the dynamic nature of the class factory).
2. Almost always tedious merge conflicts.
3. Abitrary numbers specified in input files are confusing, tedious, and fragile.

Storing relatively short, perhaps semi-abbreviated names, e.g. "strain_e", "stress_vm", etc. should almost always fit inside short string optimizations, but if we discover that these may still be to expensive to just call compare all over the place, we could just wrap them in a class that does a preliminary compare on a hash by overloading == or some such, *if* performance data shows that is it necessary.

Instead of hardcoding a huge InternalStateValueType-switch-case function, we can instead have the respective "giveIPValue" return this value, as the material model must always know what it is doing.

In addition, I think we should shorten names slightly to, e.g:  InternalVariable and VariableType

It would probably still be a good idea to predefine a large set of the most common variables, in a suitable namespace (mimicing the old enum).

**Comment by bpatzak** [comment]
I think we should have the unified "name" or "id" from controlled, common vocabulary. Not sure if I understood the description correctly. Is this just on allowing to have string representation on input? Then I am fine with that. But internally, we should map or convert these strings into common vocabulary of IDs, otherwise we can have redundant entries for the same InternalVariable.

**Comment by Micket** [comment]
I agree with your entire statement (which is what I tried to say in the last line). The most difficult part is to find the suitable names; but we should predefine all of the currently used variables (so still a huge file, like now), but it would allow externally linked material models etc. to define their new variable outside, and it would avoid the merge breaking the numbering (though, it's possible to have a name conflict instead, but these would hopefully be much rarer):
```c++
namespace InternalVariable {
    stress = "s"
    firstPKStress = "P"
    elasticStrain = "e_e"
    plasticStrain = "e_p"
    crackNormal = "crack_normal"
    ...
}
```
(it might be futile to try to use short names, except for a few common terms, most of them probably need longer names, we also might not want to use case sensitivity)

I definitely think we should handle "temp" values completely separately
